types/events.go

```diff
@@ -75,9 +76,10 @@ type EventDataNewBlockHeader struct {
 }
 
 type EventDataNewBlockEvents struct {
-	Height int64        `json:"height"`
-	Events []abci.Event `json:"events"`
-	NumTxs int64        `json:"num_txs,string"` // Number of txs in a block
+	Height       int64        `json:"height"`
+	Events       []abci.Event `json:"events"`
+	NumTxs       int64        `json:"num_txs,string"` // Number of txs in a block
+	RetainHeight int64        `json:"retain_height"`  // TODO consider another place
```

Comment:
I know it complicates things even more, but ABCI responses and the indexed data are not subject to the same retention policy. ABCI responses are stored in the blockstore if `DiscardABCIResponses = true` . Events are indexed always upon firing `EventDataNewBlockEvents`. So I am not convinced we should have the same retain height for both.

Reply:
Yes, actually, you're right.

The block retain height, however, will act as the minimum possible value for both the tx and block indexes would it not? Since when using `/tx_search` we need to load the relevant block(s) here: https://github.com/cometbft/cometbft/blob/24ebdd9df78d15828aa758e16a21d82fadcbb95b/rpc/core/tx.go#L116

And when using `/block_search` we need to load the relevant block(s) here: https://github.com/cometbft/cometbft/blob/24ebdd9df78d15828aa758e16a21d82fadcbb95b/rpc/core/blocks.go#L253

Does it then make sense really to prune indexes above the block retain height? I wouldn't think so - in that case it's better to just turn the indexers off.

Reply:
Hmm, so technically, even now, the blocks can be pruned and the `tx_search` would still have the indexing data for those heights. If you remember we tried to reason about this when we initially talked about pruning. And then I think some of our users said that even if they prune blocks they keep the indexer. Now I am not sure what scenario that could be for: Maybe use the indexer to see whether there is a match and then they have the blocks on special archive nodes where they fetch them from (not from the node which was queried via `tx_search`. This might be something we can ask Adi to find out?

Reply:
So since we'll be offering folks the ability to index their data outside of Comet, does it really make sense to keep the tx index if the related blocks have been pruned?

cc @adizere